doc/nextml2015/presentation.tex

```diff
@@ -0,0 +1,1186 @@
+                    outputs=[results],
+                    updates=updates)
+x = np.eye(10, 2, dtype=theano.config.floatX)
+w = np.ones((2, 2), dtype=theano.config.floatX)
+b = np.ones((2), dtype=theano.config.floatX)
+
+print f(x, w, b)
+\end{lstlisting}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{Scan Example2: Computing pow(A, k)}
+
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        }
+\begin{lstlisting}
+import theano
+import theano.tensor as T
+theano.config.warn.subtensor_merge_bug = False
+
+k = T.iscalar("k")
+A = T.vector("A")
+
+def inner_fct(prior_result, B):
+    return prior_result * B
+\end{lstlisting}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{Scan Example2: Computing pow(A, k) (2)}
+
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        }
+\begin{lstlisting}
+result, updates = theano.scan(
+    fn=inner_fct,
+    outputs_info=T.ones_like(A),
+    non_sequences=A, n_steps=k)
+
+# Scan provide us with A ** 1 through A ** k.
+# Keep only the last value. Scan optimize memory.
+final = result[-1]
+
+power = theano.function(inputs=[A, k], outputs=final,
+                      updates=updates)
+print power(range(10), 2)
+#[  0.   1.   4.   9.  16.  25.  36.  49.  64.  81.]
+\end{lstlisting}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{Scan signature}
+
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        }
+\begin{lstlisting}
+result, updates = theano.scan(
+    fn=inner_fct,
+    sequences=[]
+    outputs_info=[T.ones_like(A)],
+    non_sequences=A,
+    n_steps=k)
+\end{lstlisting}
+
+\begin{itemize}
+\item Updates are needed if there are random numbers generated in the inner function
+\begin{itemize}
+\item Pass them to the call theano.function(..., updates=updates)
+\end{itemize}
+\item The inner function of scan takes arguments like this:
+   scan: sequences, outputs\_info, non sequences
+\end{itemize}
+
+\end{frame}
+
+
+\section{RNN}
+\begin{frame}
+  \tableofcontents[currentsection]
+\end{frame}
+
+\begin{frame}
+  \frametitle{Recurrent Neural Network Overview}
+\begin{itemize}
+\item RNN is a class of neural network that allows to work with sequences of variable sizes.
+\item It does so, by reusing weights for each element of the sequence.
+\item It creates an internal state that allows to exhibit dynamic temporal behavior.
+\end{itemize}
+Image from wikipedia by Fyedernoggersnodden
+\includegraphics[width=0.35\textwidth]{../images/Elman_srnn.png}
+\end{frame}
+
+\section{LSTM}
+\begin{frame}
+  \tableofcontents[currentsection]
+\end{frame}
+
+\begin{frame}
+  \frametitle{Motivation}
+RNN gradient signal end up being multiplied a large number of times (as many as the number of timesteps).
+This means that, the magnitude of the weights in the transition matrix can have a strong impact on the learning process.
```

Comment:
I think the introduction on RNNs in the previous slide is not enough for people to understand this. Either add more material to the RNN intro or simplify this motivation. Maybe you could simply put it as  "The recurrent connection(s) mean that the effective depth of the model is much larger than the number of layers. This means that it will encounter the same problems typical of very deep neural networks; vanishing and exploding gradients."